src/odbc/src/app/application_data_buffer.cpp

```diff
@@ -326,14 +326,13 @@ ConversionResult::Type ApplicationDataBuffer::PutStrToStrBuffer(
     *resLenPtr = remainingBytesRequired;
   }
 
-  if (cellOffset >= 0) {
-    size_t numCharsWritten = bytesWritten / outCharSize;
-    SetCellOffset(cellOffset + numCharsWritten * inCharSize);
-  }
-
   if (isTruncated) {
+    SetCellOffset(currentCellOffset + (buflen / outCharSize));
     return ConversionResult::Type::AI_VARLEN_DATA_TRUNCATED;
   } else {
+    if (cellOffset >= 0) {
+      SetCellOffset(totalBytesWritten);
```

Comment:
Just want to make sure I understand this correctly. This sets the cell offset in bytes right? And totalBytesWritten does _not_ include bytes for the NUL terminator.

Reply:
Correct, it does not include the bytes for the NUL terminator.